test/extended/prometheus/prometheus.go

```diff
@@ -396,11 +406,20 @@ var _ = g.Describe("[sig-instrumentation] Prometheus", func() {
 				oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
 			}()
 
+			// Checking Watchdog alert state is done in "should have a Watchdog alert in firing state".
+			allowedAlerts := []string{"Watchdog", "AlertmanagerReceiversNotConfigured", "PrometheusRemoteWriteDesiredShards"}
+
+			infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if infra.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode {
```

Comment:
move these up to the top and give them unique names “allowedAlertsDuringX”, don’t reuse the same variable

Reply:
By top do you mean the top of the `It` block?

Also not sure how I can avoid reusing the same variable without duplicating the Prometheus query's `fmt.Sprintf` - it has to use some variable for `strings.Join(allowedAlerts, "|")`

Reply:
Nm, I thought you were fixing the upgrade job which has both of these inline.

Reply:
Move the infra config lookup up above allowedAlerts, then init allowedAlerts, then do the if check.  Roughly keep infra lookup high and the init of allowedAlerts and the conditional close to each other.

Reply:
Got it, done

/hold until I verify this works (takes a while)

Reply:
/unhold